Test/Tasty/Runners/AntXML.hs

```diff
@@ -177,6 +181,18 @@ antXMLRunner = Tasty.TestReporter optionDescription runner
 
           return (getSum ((summaryFailures `mappend` summaryErrors) summary) == 0)
 
+  -- Drops ANSI control characters which might be used to set colors.
+  -- Including these breaks XML, there is not much point encoding them.
+  dropConsoleFormatting input =
+    let regex = Regex.makeRegex "\x1b\\[[0-9;]*[mGKHF]" :: Regex.Regex
+        matches = Regex.matchAll regex input
+        dropMatch (offset, len) input' =
+          let (begining, rest) = splitAt offset input'
+              (_, end) = splitAt len rest
+          in begining <> end
```

Comment:
This is quadratic in the worst case, right? One could implement `dropConsoleFormatting` as a linear pass, the state machine is simple in this case.